diff --git a/package/boot/uboot-rockchip/Makefile b/package/boot/uboot-rockchip/Makefile
index ad7394c39e..c1a23bdb19 100644
--- a/package/boot/uboot-rockchip/Makefile
+++ b/package/boot/uboot-rockchip/Makefile
@@ -212,6 +212,13 @@ define U-Boot/sige3-rk3568
     armsom_sige3
 endef
 
+define U-Boot/opc-h68k-rk3568
+  $(U-Boot/rk3568/Default)
+  NAME:=OPC-H68K
+  BUILD_DEVICES:= \
+    hinlink_opc-h68k
+endef
+
 UBOOT_TARGETS := \
   guangmiao-g4c-rk3399 \
   nanopc-t4-rk3399 \
@@ -230,6 +237,7 @@ UBOOT_TARGETS := \
   fastrhino-r66s-rk3568 \
   fastrhino-r68s-rk3568 \
   lyt-t68m-rk3568 \
+  opc-h68k-rk3568 \
   mrkaio-m68s-rk3568 \
   nanopi-r5c-rk3568 \
   nanopi-r5s-rk3568 \
diff --git a/package/boot/uboot-rockchip/patches/900-arm-add-dts-files.patch b/package/boot/uboot-rockchip/patches/900-arm-add-dts-files.patch
index e5e92d6df0..74b04b4dbd 100644
--- a/package/boot/uboot-rockchip/patches/900-arm-add-dts-files.patch
+++ b/package/boot/uboot-rockchip/patches/900-arm-add-dts-files.patch
@@ -28,7 +28,7 @@
  	rk3399-orangepi.dtb \
  	rk3399-pinebook-pro.dtb \
  	rk3399-pinephone-pro.dtb \
-@@ -168,7 +174,16 @@ dtb-$(CONFIG_ROCKCHIP_RK3399) += \
+@@ -168,7 +174,17 @@ dtb-$(CONFIG_ROCKCHIP_RK3399) += \
  dtb-$(CONFIG_ROCKCHIP_RK3568) += \
  	rk3566-anbernic-rgxx3.dtb \
  	rk3566-radxa-cm3-io.dtb \
@@ -37,6 +37,7 @@
 +	rk3568-fastrhino-r66s.dtb \
 +	rk3568-fastrhino-r68s.dtb \
 +	rk3568-lyt-t68m.dtb \
++	rk3568-opc-h68k.dtb \
 +	rk3568-mrkaio-m68s.dtb \
 +	rk3568-nanopi-r5c.dtb \
 +	rk3568-nanopi-r5s.dtb \
diff --git a/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds b/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
index a5d87e5a22..0568309d52 100644
--- a/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
+++ b/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
@@ -29,6 +29,9 @@ friendlyarm,nanopi-r5s)
 	ucidef_set_led_netdev "lan1" "LAN1" "green:lan1" "eth1"
 	ucidef_set_led_netdev "lan2" "LAN2" "green:lan2" "eth2"
 	;;
+hinlink,opc-h68k)
+	ucidef_set_led_netdev "wan" "WAN" "blue:net" "eth0"
+	;;
 esac
 
 board_config_flush
diff --git a/target/linux/rockchip/armv8/base-files/etc/board.d/02_network b/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
index 33c9640c48..846ce092c5 100644
--- a/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
+++ b/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
@@ -35,6 +35,9 @@ rockchip_setup_interfaces()
 	lyt,t68m)
 		ucidef_set_interfaces_lan_wan 'lan2 lan3 lan4' 'lan1'
 		;;
+	hinlink,opc-h68k)
+		ucidef_set_interfaces_lan_wan "eth1 eth2 eth3" "eth0"
+		;;
 	*)
 		ucidef_set_interface_lan 'eth0'
 		;;
@@ -54,7 +57,8 @@ rockchip_setup_macs()
 	friendlyarm,nanopi-r2c|\
 	friendlyarm,nanopi-r2s|\
 	lunzn,fastrhino-r66s|\
-	lunzn,fastrhino-r68s)
+	lunzn,fastrhino-r68s)|\
+	hinlink,opc-h68k)
 		wan_mac=$(macaddr_generate_from_mmc_cid mmcblk0)
 		lan_mac=$(macaddr_add "$wan_mac" 1)
 		;;
diff --git a/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity b/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity
index 141d40db27..2ece6db63e 100644
--- a/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity
+++ b/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity
@@ -70,4 +70,34 @@ lyt,t68m)
 	set_interface_core 4 "lan3"
 	set_interface_core 8 "lan4"
 	;;
+hinlink,opc-h68k)
+	set_interface_core "0-3" "eth0"
+	set_interface_core "1" "eth1"
+	set_interface_core "2" "eth2-0"
+	set_interface_core "2" "eth2-16"
+	set_interface_core "1" "eth2-18"
+	set_interface_core "3" "eth3-0"
+	set_interface_core "3" "eth3-18"
+	set_interface_core "1" "eth3-16"
+	rfc=32768
+	sysctl net.core.rps_sock_flow_entries=$rfc
+	for fileRfc in $(ls /sys/class/net/eth*/queues/rx-*/rps_flow_cnt)
+	do
+		eth_name=$(echo "$fileRfc"|awk -F/ '{print $5}')
+		echo $rfc > $fileRfc
+	done
+	#0101
+	echo 9 > /sys/class/net/eth0/queues/rx-0/rps_cpus
+	echo 9 > /sys/class/net/eth1/queues/rx-0/rps_cpus
+	#0011
+	echo 3 > /sys/class/net/eth2/queues/rx-0/rps_cpus
+	echo 3 > /sys/class/net/eth2/queues/rx-1/rps_cpus
+	#1100
+	echo c > /sys/class/net/eth3/queues/rx-0/rps_cpus
+	echo c > /sys/class/net/eth3/queues/rx-1/rps_cpus
+	/usr/sbin/ethtool -K eth0 tso on sg on tx on
+	/usr/sbin/ethtool -K eth1 tso on sg on tx on
+	/usr/sbin/ethtool -K eth2 tso on sg on tx on
+	/usr/sbin/ethtool -K eth3 tso on sg on tx on
+	;;
 esac
diff --git a/target/linux/rockchip/image/armv8.mk b/target/linux/rockchip/image/armv8.mk
index ffe67b0377..35f9859e43 100644
--- a/target/linux/rockchip/image/armv8.mk
+++ b/target/linux/rockchip/image/armv8.mk
@@ -237,3 +237,16 @@ define Device/xunlong_orangepi-r1-plus-lts
   DEVICE_PACKAGES := kmod-usb-net-rtl8152
 endef
 TARGET_DEVICES += xunlong_orangepi-r1-plus-lts
+
+define Device/hinlink_common
+  DEVICE_VENDOR := HINLINK
+  BOOT_FLOW := pine64-img
+  DEVICE_PACKAGES := kmod-ata-ahci-platform kmod-hwmon-pwmfan kmod-mt7921e kmod-r8125 wpad-openssl luci-app-modem
+endef
+
+define Device/hinlink_opc-h68k
+$(call Device/hinlink_common)
+  DEVICE_MODEL := OPC-H68K
+  SOC := rk3568
+endef
+TARGET_DEVICES += hinlink_opc-h68k
diff --git a/target/linux/rockchip/patches-5.15/900-arm64-boot-add-dts-files.patch b/target/linux/rockchip/patches-5.15/900-arm64-boot-add-dts-files.patch
index 7f286849fd..3bf85b706c 100644
--- a/target/linux/rockchip/patches-5.15/900-arm64-boot-add-dts-files.patch
+++ b/target/linux/rockchip/patches-5.15/900-arm64-boot-add-dts-files.patch
@@ -16,7 +16,7 @@
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-nanopi-r4s-enterprise.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-orangepi.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-pinebook-pro.dtb
-@@ -56,9 +58,13 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-ro
+@@ -56,9 +58,14 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-ro
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire-excavator.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399pro-rock-pi-n10.dtb
@@ -25,6 +25,7 @@
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-fastrhino-r66s.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-fastrhino-r68s.dtb
 +dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-lyt-t68m.dtb
++dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-opc-h68k.dtb
 +dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-mrkaio-m68s.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-nanopi-r5c.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-nanopi-r5s.dtb
